common/src/main/java/com/dfsek/terra/api/util/collections/ProbabilityCollection.java

```diff
@@ -2,24 +2,28 @@
 
 import com.dfsek.terra.api.math.MathUtil;
 import com.dfsek.terra.api.math.noise.NoiseSampler;
+import com.dfsek.terra.api.util.mutable.MutableInteger;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
+import java.util.Map;
 import java.util.Random;
 import java.util.Set;
+import java.util.function.Function;
 
 @SuppressWarnings("unchecked")
 public class ProbabilityCollection<E> implements Collection<E> {
-    private final Set<E> cont = new HashSet<>();
+    protected final Map<E, MutableInteger> cont = new HashMap<>();
```

Comment:
The map is only used during additions, and operations like fetching probabilities. It is very similar in performance for those operations. Gets are identical in performance. Memory will be ever so slightly larger, the difference is negligible, though, the utility of using a map here is well worth the tiny difference.

Reply:
Aight, just checking. I haven't done any test on perf of a HashMap vs a HashSet at any point.

Actually,,, I think the memory might be slightly lower since iirc the HashSet uses a HashMap backend.